### PR TITLE
Fix `String::simplify_path` handling of relative paths to parent dir (`../`), fixes relative shader includes

### LINKED ISSUE #95016: relative #include in `.gdshader`/`.gdshaderinc` file doesn't work when using `..`
### Tested versions

v4.2.2.stable.official [15073afe3]

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated AMD Radeon RX 6700 XT (Advanced Micro Devices, Inc.; 31.0.23013.3005) - AMD Ryzen 7 2700 Eight-Core Processor (16 Threads)

### Issue description

When including a shader in a parent folder using a relative path, the shader include cannot be found.

example structure:

```plain
shaders
  test
    include1.gdshaderinc
    main.gdshader
  include2.gdshaderinc
```

### Steps to reproduce

`shaders/test/main.gdshader`

```glsl
shader_type spatial;
#include "./include1.gdshaderinc"
```

`shaders/test/include1.gdshaderinc`

```glsl
#include "../include2.gdshaderinc"
```

`shaders/include2.gdshader`

```glsl
bool do_something() { return true; }
```

Error message:

```
error(1): Shader include file does not exist: res://shaders/test/include2.gdshaderinc
```

### Minimal reproduction project (MRP)

[shader-incl-error.zip](https://github.com/user-attachments/files/16456505/shader-incl-error.zip)

### Work around

To get around this problem, you can use `res://shaders/include2.gdshader`, however that doesn't work if this is a shared asset as it is unknown where the files will be saved in the project.

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/95016

### CODE REVIEW

**Comment by akien-mga** [comment]
I would argue that it's `String::simplify_path` which is broken. `../somefile` shouldn't be simplified to `somefile`.

The function is meant to help with cases like `path/sub/../file` to simplify it to `path/file`, but it shouldn't remove a leading `../` if there's no folder path before it.

**Comment by jsjtxietian** [comment]
> I would argue that it's `String::simplify_path` which is broken. `../somefile` shouldn't be simplified to `somefile`.

Indeed. I will  take a look at the function more closely.

File: core/string/ustring.cpp [hunk 1]
```diff
@@ -4378,7 +4378,7 @@ String String::simplify_path() const {
 	}
 	Vector<String> dirs = s.split("/", false);
 
-	for (int i = 0; i < dirs.size(); i++) {
+	for (int i = 1; i < dirs.size(); i++) {
```
Comment (akien-mga) [inline core/string/ustring.cpp:4381]: This is wrong, you can see that the `for` loop explicitly handles `i == 0` below.

And a leading `./` needs to be removed.

What should be done here is remove the `i == 0` branch in `d == ".."`.

And adjust the tests accordingly as they seem to validate the wrong behavior.

Comment (jsjtxietian) [reply core/string/ustring.cpp]: Done

File: tests/core/config/test_project_settings.h [hunk 2]
```diff
@@ -127,9 +127,9 @@ TEST_CASE("[ProjectSettings] localize_path") {
 #endif
 
 	// FIXME?: These checks pass, but that doesn't seems correct
```
Comment (akien-mga) [inline tests/core/config/test_project_settings.h:129]: You fixed this :)

File: core/string/ustring.cpp (hunk 1 above)
Comment (akien-mga) [reply core/string/ustring.cpp]: Yes you also need to fix that test.
```diff
diff --git a/tests/core/string/test_string.h b/tests/core/string/test_string.h
index cf57183a02..1b27966da1 100644
--- a/tests/core/string/test_string.h
+++ b/tests/core/string/test_string.h
@@ -1594,7 +1594,7 @@ TEST_CASE("[String] Path functions") {
 	static const char *base_name[8] = { "C:\\Godot\\project\\test", "/Godot/project/test", "../Godot/project/test", "Godot\\test", "C:\\test", "res://test", "user://test", "/" };
 	static const char *ext[8] = { "tscn", "xscn", "scn", "doc", "", "", "", "test" };
 	static const char *file[8] = { "test.tscn", "test.xscn", "test.scn", "test.doc", "test.", "test", "test", ".test" };
-	static const char *simplified[8] = { "C:/Godot/project/test.tscn", "/Godot/project/test.xscn", "Godot/project/test.scn", "Godot/test.doc", "C:/test.", "res://test", "user://test", "/.test" };
+	static const char *simplified[8] = { "C:/Godot/project/test.tscn", "/Godot/project/test.xscn", "../Godot/project/test.scn", "Godot/test.doc", "C:/test.", "res://test", "user://test", "/.test" };
 	static const bool abs[8] = { true, true, false, false, true, true, true, true };
 
 	for (int i = 0; i < 8; i++) {
```

File: tests/core/config/test_project_settings.h (hunk 2 above)
Comment (jsjtxietian) [reply tests/core/config/test_project_settings.h:129]: lol, I will remove the comment.

File: core/string/ustring.cpp (hunk 1 above)
Comment (jsjtxietian) [reply core/string/ustring.cpp]: Thanks!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**